client/src/api/friendsApi.js

```diff
@@ -0,0 +1,150 @@
+/**
+ *
+ * GET Requests
+ *
+ *   */
+
+async function getFriends() {
+  try {
+    return await makeGetRequest('/api/friends');
+    // axios.defaults.withCredentials = true;
+    // const response = await axios.get('/api/friends');
+    // return response;
+  } catch (err) {
+    return err;
+  }
+}
+
+async function getSuggestedFriends() {
+  try {
+    return await makeGetRequest('/api/suggested-friends');
+  } catch (err) {
+    return err;
+  }
+}
+
+async function getReceivedRequests() {
+  try {
+    return await makeGetRequest('/api/received-requests');
+  } catch (err) {
+    return err;
+  }
+}
+
+/**
+ *
+ *  POST Requests
+ *
+ *  */
+async function postNewFriendRequest(friendId) {
+  try {
+    return await makePostRequest('/api/outgoing-requests', friendId);
+  } catch (err) {
+    return err;
+  }
+}
+
+async function postAcceptFriendRequest(friendId) {
+  try {
+    return await makePostRequest('/api/friends', friendId);
+  } catch (err) {
+    return err;
+  }
+}
+
+/**
+ *
+ * DELETE Requests
+ *
+ *
+ */
+async function deleteOutgoingFriendRequest(friendId) {
+  try {
+    return await makeDeleteRequest(`/api/outgoing-requests/${friendId}`);
+  } catch (err) {
+    return err;
+  }
+}
+async function deleteFriend(friendId) {
+  try {
+    return await makeDeleteRequest(`/api/friends/${friendId}`);
+  } catch (err) {
+    return err;
+  }
+}
+async function deleteReceivedFriendRequest(friendId) {
+  try {
+    return await makeDeleteRequest(`/api/received-requests/${friendId}`);
+  } catch (err) {
+    return err;
+  }
+}
+
+/**
+ *
+ * Helper functions
+ *
+ *
+ */
+async function makeGetRequest(url) {
```

Comment:
not sure why these codes reflect on your branch should not. I see many lines of codes written by me most importantly i delete a few of them but wondering why it exists in your branch... is it an old branch ??

Reply:
I merged your in-create-polls branch so I have the code that you wrote on that branch. Which one is your latest one? I can pull that one and merge it then